### ISSUE TITLE
Firehose does not shut down properly

### ISSUE THREAD

**Issue by lukinhas563** [MarshalX/atproto#383]
When you use the Firehose and try to shut it down later, the process doesn't work as expected. It blocks the flow and doesn't proceed to the next operation. You can stop it with `client.stop()`, but you can't shut it down.

You can observe this issue in the code below `(stop_client_async.py)`:

```python
import asyncio

from atproto import (
    AsyncFirehoseSubscribeReposClient,
    firehose_models,
    parse_subscribe_repos_message,
)

_STOP_AFTER_SECONDS = 3


async def main() -> None:
    client = AsyncFirehoseSubscribeReposClient()

    async def on_message_handler(message: firehose_models.MessageFrame) -> None:
        print(message.header, parse_subscribe_repos_message(message))

    async def _stop_after_n_sec() -> None:
        await asyncio.sleep(_STOP_AFTER_SECONDS)
        await client.stop()
        print('Stopped')

    # save ref to task to eliminate problems with GC
    _ = asyncio.create_task(_stop_after_n_sec())

    await client.start(on_message_handler)

    print(f'Successfully stopped after {_STOP_AFTER_SECONDS} seconds!')


if __name__ == '__main__':
    # use run() for a higher Python version
    asyncio.get_event_loop().run_until_complete(main())
```

As you can see, the `print` statement, after `client.stop()`, never gets executed.

I am using version 0.0.52

**Comment by MarshalX** [comment]
Could you try to downgrade to 0.0.51? Same?

**Comment by lukinhas563** [comment]
I downgraded to 0.0.51, and it works fine. I'll stick with that version for now. Thanks for the help.

**Comment by MarshalX** [comment]
@lukinhas563 pls verify fix. fixed in 0.0.53
